portal-impl/src/com/liferay/portal/zip/ZipWriterImpl.java

```diff
@@ -119,6 +120,18 @@ public void addEntry(String name, InputStream inputStream)
 			return;
 		}
 
+		if (ExportImportThreadLocal.isExportInProcess()) {
```

Comment:
@mbowerman If you compare the logic inside` addEntry(String name, byte[] bytes)` and this method, you would see they have similar code. Could you please try to avoid duplicated logic? My prefer would be: read from upstream to get the byte array, then pass it into ` addEntry(String name, byte[] bytes)` if there is no side effect by doing this.

Thanks.
Tina.